main/mafia.hs

```diff
@@ -613,3 +625,68 @@ sandbox_ :: Argument -> [Argument] -> EitherT MafiaViolation IO ()
 sandbox_ cmd args = do
   Pass <- sandbox cmd args
   return ()
+
+------------------------------------------------------------------------
+
+getMafiaHome :: IO Text
+getMafiaHome =
+  (</> T.pack ".ambiata/mafia") <$> getHomeDirectory
+
+ensureMafiaDir :: MonadIO m => Text -> m Text
+ensureMafiaDir path = liftIO $ do
+  home <- getMafiaHome
+  let path' = home </> path
+  createDirectoryIfMissing True path'
+  pure path'
+
+hoogle :: Text -> [Argument] -> EitherT MafiaViolation IO ()
+hoogle hackageRoot args = do
+  initialize
+  db <- ensureMafiaDir "hoogle"
+  hoogleExe <- installBinary "hoogle" "4.2.43" [("happy", "1.19.5")]
+  Out pkgStr <- sandbox "hc-pkg" ["list"]
+  let pkgs = fmap T.strip . filter (T.isPrefixOf " ") . T.lines $ pkgStr
+  hoos <- for pkgs $ \pkg -> do
+    let name = T.intercalate "-" . List.init $ T.splitOn "-" pkg
+    let txt = db </> pkg <> ".txt"
+    let hoo = db </> pkg <> ".hoo"
+    unlessM (doesFileExist txt) $ do
```

Comment:
It's an ambiata `P` special
On 14 Dec 2015 7:28 PM, "Jacob Stanley" <notifications@github.com> wrote:

> In main/mafia.hs
> <https://github.com/ambiata/mafia/pull/26#discussion_r47470817>:
>
> > +  let path' = home </> path
> > +  createDirectoryIfMissing True path'
> > +  pure path'
> > +
> > +hoogle :: Text -> [Argument] -> EitherT MafiaViolation IO ()
> > +hoogle hackageRoot args = do
> > +  initialize
> > +  db <- ensureMafiaDir "hoogle"
> > +  hoogleExe <- installBinary "hoogle" "4.2.43" [("happy", "1.19.5")]
> > +  Out pkgStr <- sandbox "hc-pkg" ["list"]
> > +  let pkgs = fmap T.strip . filter (T.isPrefixOf " ") . T.lines $ pkgStr
> > +  hoos <- for pkgs $ \pkg -> do
> > +    let name = T.intercalate "-" . List.init $ T.splitOn "-" pkg
> > +    let txt = db </> pkg <> ".txt"
> > +    let hoo = db </> pkg <> ".hoo"
> > +    unlessM (doesFileExist txt) $ do
>
> nice, i didn't know about unlessM
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ambiata/mafia/pull/26/files#r47470817>.
>